File: src/main/java/com/paragon/client/systems/module/impl/misc/RotationLock.java
package com.paragon.client.systems.module.impl.misc;

import com.paragon.api.module.Category;
import com.paragon.api.module.Module;
import com.paragon.api.setting.Setting;

/**
 * @author Surge
 */
public class RotationLock extends Module {

    public static RotationLock INSTANCE;

    public static Setting<Float> yaw = new Setting<>("Yaw", 0f, -180f, 180f, 1f)
            .setDescription("The yaw to lock to");

    public static Setting<Float> pitch = new Setting<>("Pitch", 0f, -180f, 180f, 1f)
            .setDescription("The pitch to lock to");

    public RotationLock() {
        super("RotationLock", Category.MISC, "Locks your rotation");

        INSTANCE = this;
    }

    @Override
    public void onTick() {
        if (nullCheck()) {
            return;
        }

        mc.player.rotationYaw = yaw.getValue();
        mc.player.rotationYawHead = yaw.getValue();
        mc.player.rotationPitch = pitch.getValue();
    }
}

File: src/main/java/com/paragon/client/systems/module/impl/misc/RotationLock.kt
package com.paragon.client.systems.module.impl.misc

import com.paragon.api.module.Category
import com.paragon.api.module.Module
import com.paragon.api.setting.Setting
import com.paragon.api.util.anyNull

/**
 * @author Surge
 */
object RotationLock : Module("RotationLock", Category.MISC, "Locks your rotation") {

    private val yaw = Setting(
        "Yaw",
        0f,
        -180f,
        180f,
        1f
    ) describedBy "The yaw to lock to"

    private val pitch = Setting(
        "Pitch",
        0f,
        -180f,
        180f,
        1f
    ) describedBy "The pitch to lock to"

    override fun onTick() {
        if (minecraft.anyNull) {
            return
        }

        minecraft.player.rotationYaw = yaw.value
        minecraft.player.rotationYawHead = yaw.value
        minecraft.player.rotationPitch = pitch.value
    }

}